### PR TITLE
Fix test to use PublishNotReadyAddresses instead of deprecated annotation

### LINKED ISSUE #78834: Write Scale Tests for Conformance
Using the list of tested stable endpoints referenced in https://github.com/kubernetes/kubernetes/issues/78748, we can pick out the endpoints which are part of the Scale kind:

[replaceCoreV1NamespacedReplicationControllerScale](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-replaceCoreV1NamespacedReplicationControllerScale&showUntested=false&showConformanceTested=false)

[readCoreV1NamespacedReplicationControllerScale](https://apisnoop.cncf.io/?zoomed=operationId-stable-core-readCoreV1NamespacedReplicationControllerScale&showUntested=false&showConformanceTested=false)

These are stable core endpoints which are covered by an existing test, but the test does not directly exercise these endpoints. New tests should be written.

/area test
/area conformance
/sig testing

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fixes and updates test as to [these comments](https://github.com/kubernetes/kubernetes/pull/79122#issuecomment-506766182)

**Which issue(s) this PR fixes**:
Fixes #78834

**Special notes for your reviewer**:
Part of Umbrella Issue #78748

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/area test

@kubernetes/sig-architecture-pr-reviews
@kubernetes/sig-network-pr-reviews

### CODE REVIEW

**Comment by devaips** [comment]
@johnbelamaric this PR is an attempt to rewrite a service test following [your comments](https://github.com/kubernetes/kubernetes/pull/79122#issuecomment-506766182) on a [previous promotion PR](https://github.com/kubernetes/kubernetes/pull/79122).

Currently the tests are failing. See @BobyMCbobs's comment on this here: https://github.com/kubernetes/kubernetes/pull/79122#issuecomment-518856820

**Comment by johnbelamaric** [comment]
Did you forget a file in the commits?

```
/b/f/w/test/e2e/network/service.go:1395:12: undefined: framework.UpdateService
/b/f/w/test/e2e/network/service.go:1415:12: undefined: framework.UpdateService
```

**Comment by devaips** [comment]
@bowei thanks for assigning yourself. Can you verify whether this test including the proposed changes from https://github.com/kubernetes/kubernetes/pull/79122#issuecomment-506766182 would be sound for the behaviors which it is trying to exercise.

**Comment by BobyMCbobs** [comment]
> Did you forget a file in the commits?
> 
> ```
> /b/f/w/test/e2e/network/service.go:1395:12: undefined: framework.UpdateService
> /b/f/w/test/e2e/network/service.go:1415:12: undefined: framework.UpdateService
> ```

It appears that the given lines were updated automatically and unintentionally.
These calls have been reverted back in 7e2a975d0dcaafd958413d65f679c54f4a702855.

**Comment by bowei** [comment]
Is this still WIP or is this PR ready for review?

**Comment by BobyMCbobs** [comment]
> Is this still WIP or is this PR ready for review?

Hey @bowei, it's ready for review now.

**Comment by alejandrox1** [comment]
@BobyMCbobs could you please run the following command 
```
./hack/update-bazel.sh
```
This should make the tests pass

**Comment by BobyMCbobs** [comment]
> @BobyMCbobs could you please run the following command
> 
> ```
> ./hack/update-bazel.sh
> ```
> 
> This should make the tests pass

Pushed changes from this in https://github.com/kubernetes/kubernetes/pull/81320/commits/a3771ccb3b1611833092e597ef5835a6d7e2f5bd

**Comment by johnbelamaric** [comment]
/retitle "Fix test to use PublishNotReadyAddresses instead of deprecated annotation"

**Comment by johnbelamaric** [comment]
/retitle Fix test to use PublishNotReadyAddresses instead of deprecated annotation

**Comment by smourapina** [comment]
@johnbelamaric @BobyMCbobs Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on November 18. Is this PR still intended for 1.17?

**Comment by BobyMCbobs** [comment]
> @johnbelamaric @BobyMCbobs Bug triage for 1.17 here with a gentle reminder that code freeze for this release is on November 18. Is this PR still intended for 1.17?

1.17 would be great if possible, otherwise 1.18 is fine

**Comment by johnbelamaric** [comment]
@BobyMCbobs looks like these are consistently failing, can you dig into this?

**Comment by danwinship** [comment]
ping @bowei

**Comment by spiffxp** [comment]
/retest
can you squash commits and ensure all authors/committers have signed the CLA to take care of the CLA issue?

File: test/e2e/network/service.go [hunk 1]
```diff
@@ -1978,13 +1978,6 @@ var _ = SIGDescribe("Services", func() {
 					},
 				},
 			},
-			Lifecycle: &v1.Lifecycle{
```
Comment (spiffxp) [inline test/e2e/network/service.go]: I think the deletion of this is causing the test to fail

Comment (johnbelamaric) [reply test/e2e/network/service.go]: if it's using the proper agnhost image and command then this shoudn't be needed, I don't think?

Comment (BobyMCbobs) [reply test/e2e/network/service.go]: That's what was [considered](https://github.com/kubernetes/kubernetes/pull/79122#issuecomment-518856820), I'll try and implement this

**Comment by BobyMCbobs** [comment]
I've updated the test to use a new agnhost container to try and connect to the service to determine if the test service is live instead of using a host based cURL command. The test will likely fail but there's been some progress made.

**Comment by spiffxp** [comment]
I feel like the simpler change would be to first switch the annotation usage to the field in this PR, and try cleaning the test up as a followup, rather than trying to do both at once

**Comment by johnbelamaric** [comment]
Well, I am ok with that, but the switch to agnhost will be needed to
promote this.

**Comment by hh** [comment]
@BobyMCbobs is looking into this and will update this week.

**Comment by annajung** [comment]
Hello @johnbelamaric @BobyMCbobs   👋 I'm one of the Bug Triage shadows for the 1.18 release. Since this PR is tagged for 1.18, I'd like to check its status. Is this PR intended for milestone 1.18?

Friendly reminder: Code freeze is scheduled for March 5th.

**Comment by johnbelamaric** [comment]
So....this was already done it looks like quite some time ago by another PR. That is, it is already no longer using the deprecated annotations. There is no need to just rearrange the fields.

**Comment by annajung** [comment]
Hello @johnbelamaric, I want to double-check. Can we remove this from v1.18 release and close this PR?

**Comment by hh** [comment]
/retest
@johnbelamaric this would I think give us +2 stable conformance endpoints

**Comment by johnbelamaric** [comment]
@hh no, it doesn't do that anymore, that was done in a previous PR. This actually doesn't have any effect now - it just moves some lines around.

/close

**Pull request closed without merging**